subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/TestFrameworkAutoDetectionIntegrationTest.groovy

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.testing
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.integtests.fixtures.DefaultTestExecutionResult
+import org.gradle.testing.fixture.JUnitCoverage
+import org.gradle.testing.fixture.TestNGCoverage
+
+class TestFrameworkAutoDetectionIntegrationTest extends AbstractIntegrationSpec {
```

Comment:
The dependencies used in these tests are sort of confusing. 

In some cases, we're adding `junit-jupiter`, which appears to be the right dependency for JUnit 5+, but sometimes we're adding junit-jupiter-engine and we're almost always adding testng.  And we're mostly adding dependencies to testImplementation, but our samples use a wildly different mix.

I think these tests should assert the test framework used vs relying on just the tests passing/failing.  I think that could be done by adding a `doLast` to the test task and asserting testFramework has a particular type.

Reply:
IIRC `testng` is always there to show that both JUnit Platform and JUnit take precedence.